### PR TITLE
Move rest/admin to use async/await.

### PR DESCRIPTION
**Pull request by erikjohnston**
Based on #6195

### CODE REVIEW

File: synapse/rest/admin/__init__.py [hunk 1]
```diff
@@ -318,7 +310,7 @@ def on_POST(self, request, room_id, event_id):
                 errcode=Codes.BAD_JSON,
             )
 
-        purge_id = yield self.pagination_handler.start_purge_history(
+        purge_id = await self.pagination_handler.start_purge_history(
```
Comment (richvdh) [inline synapse/rest/admin/__init__.py]: start_purge_history appears to be synchronous

File: synapse/rest/admin/__init__.py [hunk 2]
```diff
@@ -491,9 +480,9 @@ def on_POST(self, request, room_id):
             ratelimit=False,
         )
 
-        aliases_for_room = yield self.store.get_aliases_for_room(room_id)
+        aliases_for_room = await self.store.get_aliases_for_room(room_id)
```
Comment (richvdh) [inline synapse/rest/admin/__init__.py]: get_aliases_for_room has a cache so *may* return synchronously

File: synapse/rest/admin/_base.py [hunk 3]
```diff
@@ -58,12 +55,11 @@ def assert_requester_is_admin(auth, request):
...
+    requester = await auth.get_user_by_req(request)
+    await assert_user_is_admin(auth, requester.user)
 
 
-@defer.inlineCallbacks
-def assert_user_is_admin(auth, user_id):
+async def assert_user_is_admin(auth, user_id):
     """Verify that the given user is an admin user
 
     WARNING: MAKE SURE YOU YIELD ON THE RESULT!
```
Comment (richvdh) [inline synapse/rest/admin/_base.py]: ... or await

File: synapse/util/async_helpers.py [hunk 4]
```diff
@@ -483,3 +485,30 @@ def failure_cb(val):
     deferred.addCallbacks(success_cb, failure_cb)
 
     return new_d
+
+
+@attr.s(slots=True, frozen=True)
+class DoneAwaitable(object):
```
Comment (hawkowl) [inline synapse/util/async_helpers.py]: can you not use `twisted.internet.defer.succeed`?

**Pull request merged**